### PR TITLE
Evict TCP connections after a configurable timeout

### PR DESCRIPTION
**Pull request by leeavital**
Sometimes, tcp_close calls are not intercepted by the BPF module causing
us to leave dead connections around. This will evict TCP connections
from the map after a ten minute (configurable) timeout.

### CODE REVIEW

File: tcptracer-bpf.c [hunk 1]
```diff
@@ -508,15 +513,22 @@ static int increment_tcp_stats(struct sock *sk, struct tcptracer_status_t *statu
 			if (val != NULL) {
 				(*val).send_bytes += send_bytes;
 				(*val).recv_bytes += recv_bytes;
+        (*val).timestamp = ts;
```
Comment (sunhay) [inline tcptracer-bpf.c]: Formatting looks a bit off - we should probably switch to using spaces for this entire file

Comment (leeavital) [reply tcptracer-bpf.c]: Whoops, I'm going to do a pre-commit running this through a formatter

File: tcptracer-bpf.c [hunk 2]
```diff
@@ -50,24 +50,24 @@ struct bpf_map_def SEC("maps/udp_stats_ipv6") udp_stats_ipv6 = {
 };
 
 /* This is a key/value store with the keys being an ipv4_tuple_t for send & recv calls
- * and the values being the struct conn_stats_t *.
+ * and the values being the struct conn_stats_ts_t *.
  */
 struct bpf_map_def SEC("maps/tcp_stats_ipv4") tcp_stats_ipv4 = {
```
Comment (sunhay) [inline tcptracer-bpf.c]: Perhaps for another PR but we can now merge `tcp_stats_ipv4` and `udp_stats_ipv4` maps together if we add a field to `ipv4_tuple_t` representing whether its tcp or udp

Comment (leeavital) [reply tcptracer-bpf.c]: I'm pretty sure we can. I didn't want to mix a huge refactor and this bugfix in the same PR, but can try that after this merges

**Pull request merged**